core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -233,10 +233,17 @@ class KafkaApis(val requestChannel: RequestChannel,
 
     authorizeClusterAction(request)
 
-    val partitionsRemaining = controller.shutdownBroker(controlledShutdownRequest.brokerId)
-    val controlledShutdownResponse = new ControlledShutdownResponse(controlledShutdownRequest.correlationId,
-      Errors.NONE, partitionsRemaining)
-    requestChannel.sendResponse(new Response(request, new RequestOrResponseSend(request.connectionId, controlledShutdownResponse)))
+    def controlledShutdownCallback(controlledShutdownResult: Try[Set[TopicAndPartition]]): Unit = {
+      if (controlledShutdownResult.isSuccess) {
+        val partitionsRemaining = controlledShutdownResult.get
```

Comment:
The way to handle a `Try` is to use pattern matching instead of the `isSuccess`/`get`. Generally, `get` should be seen as an unsafe method and ideally never used.